src/CarbonAware.LocationSources/src/LocationSource.cs

```diff
@@ -0,0 +1,137 @@
+using CarbonAware.Model;
+using CarbonAware.Interfaces;
+using CarbonAware.Exceptions;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using Microsoft.Extensions.Options;
+using CarbonAware.LocationSources.Configuration;
+
+namespace CarbonAware.LocationSources;
+
+/// <summary>
+/// Represents a location source.
+/// </summary>
+public class LocationSource : ILocationSource
+{
+    private readonly ILogger<LocationSource> _logger;
+
+    private IDictionary<string, NamedGeoposition>? _namedGeopositions;
```

Comment:
Any reason not to initialize this?

Reply:
> Any reason not to initialize this to make it non-nullable?

Nothing in particular. Just keeping the min changes, to have a more comprehensive diff to be read by reviewers.
Like the suggestion.